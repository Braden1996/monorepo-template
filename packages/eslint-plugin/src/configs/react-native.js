const path = require('path');

module.exports = {
  extends: [path.join(__dirname, './react')],
  plugins: ['react-native'],
  env: {
    browser: false,
    'react-native/react-native': true,
  },
  settings: {
    // Ignore non-transpiled code, often found within RN packages.
    'import/ignore': ['^react-native$'],
  },
};
