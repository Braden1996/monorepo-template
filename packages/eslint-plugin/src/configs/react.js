const path = require('path')

const {getMonorepoPackagePrefix} = require('@monorepo-template/monorepo-helpers')

const monorepoPrefix = getMonorepoPackagePrefix(__dirname);

module.exports = {
  extends: [path.join(__dirname, './default')],
  plugins: ['react', 'react-hooks'],
  env: {
    browser: true,
    es6: true,
  },
  rules: {
    'react/destructuring-assignment': 0,
    'react/jsx-filename-extension': [1, { extensions: ['.ts', '.tsx'] }],
    'react/no-array-index-key': 2,
    'react/prefer-stateless-function': 0,
    'react/prop-types': 0,
    'react/require-default-props': 0,
    'react/jsx-uses-vars': 1,
    'react/jsx-uses-react': 1,
    'react-hooks/rules-of-hooks': 'error',
    'react-hooks/exhaustive-deps': 'warn',
  },
  settings: {
    'import/internal-regex': `^(${monorepoPrefix}/|@react-test-renderer)`,
  },
};
