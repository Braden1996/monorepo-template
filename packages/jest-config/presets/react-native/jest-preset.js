const defaultPreset = require('../default/jest-preset');

// Note: due to how the 'react-native' preset is resolved by Jest, this file
// cannot be set directly as a preset - but instead must be required.
module.exports = {
  ...defaultPreset,
  preset: 'react-native',
  transformIgnorePatterns: [
    'node_modules/(?!(' +
      [
        'expo(nent)?',
        '@expo(nent)?/.*',
        'jest-expo',
        'react-native',
        'unimodules',
        '@unimodules/.*',
      ].join('|') +
      ')/)',
  ],
  modulePaths: ['<rootDir>'],
  setupFiles: ['@monorepo-template/jest-config/src/setup/native.ts'],
  setupFilesAfterEnv: [
    '@monorepo-template/jest-config/src/setupFilesAfterEnv/native.ts',
  ],
  transform: {
    '\\.[jt]sx?$': 'babel-jest',
  },
  moduleNameMapper: {
    '^react$': '<rootDir>/node_modules/react',
    '^react-native$': '<rootDir>/node_modules/react-native',
  },
  globals: {
    self: {},
  },
};
