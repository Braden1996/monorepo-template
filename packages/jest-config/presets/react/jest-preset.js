const defaultPreset = require('../default/jest-preset');

module.exports = {
  ...defaultPreset,
  testEnvironment: 'jsdom',
  transformIgnorePatterns: ['<rootDir>/node_modules', '/node_modules/*'],
  setupFiles: ['@monorepo-template/jest-config/src/setup/dom.ts'],
  setupFilesAfterEnv: [
    '@monorepo-template/jest-config/src/setupFilesAfterEnv/dom.ts',
  ],
};
