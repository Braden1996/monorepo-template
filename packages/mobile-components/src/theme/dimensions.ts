import { mapValues } from 'lodash';

const base = {
  xxSmall: 2,
  xSmall: 4,
  small: 8,
  normal: 16,
  large: 32,
  xLarge: 64,
};

const use = {
  borderRadius: {
    normal: base.xxSmall,
    large: base.xSmall,
  },
  insets: mapValues(base, value => ({
    top: value,
    right: value,
    bottom: value,
    left: value,
  })),
};

const dimensions = { base, use };

export default dimensions;
