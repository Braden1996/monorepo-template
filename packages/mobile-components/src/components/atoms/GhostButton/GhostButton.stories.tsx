import { action } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import React from 'react';

import { CenterView } from '../../../utils/story-layout/CenterVew';

import GhostButton from './index';

console.log('Hello world!');

storiesOf('GhostButton', module)
  .addDecorator(getStory => <CenterView>{getStory()}</CenterView>)
  .add('with text', () => (
    <GhostButton onPress={action('clicked-text')}>
      {text('GhostButton text', 'Hello GhostButton')}
    </GhostButton>
  ))
  .add('with some emoji', () => (
    <GhostButton onPress={action('clicked-emoji')}>😀 😎 👍 💯</GhostButton>
  ));
