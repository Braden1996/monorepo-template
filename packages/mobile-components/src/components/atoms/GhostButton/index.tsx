import * as React from 'react';
import * as ReactIs from 'react-is';
import { GestureResponderEvent, PressableProps } from 'react-native';
import styled from 'styled-components/native';

import { Pressable } from '../../particles';

const Container = styled(Pressable)`
  flex-direction: row;
  align-items: center;
  padding: ${p => p.theme.dimensions.base.small}px;
  border-radius: ${p => p.theme.dimensions.use.borderRadius.normal}px;
`;

interface ChildrenProps {
  selected: boolean;
  small: boolean;
}

const Text = styled.Text<ChildrenProps>`
  color: ${p =>
    p.selected
      ? p.theme.colors.semantic.accent.primary
      : p.theme.colors.semantic.text.primary};
`;

interface Props extends PressableProps {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactNode | React.ComponentType<ChildrenProps | any>;
  small?: boolean;
  forcePressingState?: boolean;
}

const GhostButton: React.FC<Props> = ({
  children: Children,
  onPressIn,
  onPressOut,
  small = false,
  forcePressingState = false,
  ...rest
}) => {
  const [pressing, setPressing] = React.useState(false);
  const isSelected = pressing || forcePressingState;

  const innerOnPressIn = React.useCallback(
    (evt: GestureResponderEvent) => {
      setPressing(true);
      onPressIn && onPressIn(evt);
    },
    [onPressIn],
  );

  const innerOnPressOut = React.useCallback(
    (evt: GestureResponderEvent) => {
      setPressing(false);
      onPressOut && onPressOut(evt);
    },
    [onPressOut],
  );

  return (
    <Container
      onPressIn={innerOnPressIn}
      onPressOut={innerOnPressOut}
      {...rest}
    >
      {typeof Children === 'string' || typeof Children === 'number' ? (
        <Text selected={isSelected} small={small}>
          {Children}
        </Text>
      ) : ReactIs.isValidElementType(Children) ? (
        <Children selected={isSelected} small={small} />
      ) : (
        Children
      )}
    </Container>
  );
};

export default GhostButton;
