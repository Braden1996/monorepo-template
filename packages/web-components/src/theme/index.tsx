import * as React from 'react';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';

import {
  darkSemanticTheme,
  lightSemanticTheme,
  palette,
} from '@monorepo-template/brand';

import dimensions from './dimensions';

export interface Theme {
  name: NonNullable<Props['theme']>;
  colors: {
    palette: typeof palette;
    semantic: typeof lightSemanticTheme | typeof darkSemanticTheme;
  };
  dimensions: typeof dimensions;
}

export const getTheme = (theme: 'light' | 'dark') => ({
  name: theme,
  colors: {
    palette: palette,
    semantic: theme === 'dark' ? darkSemanticTheme : lightSemanticTheme,
  },
  dimensions,
});

interface Props {
  theme?: 'light' | 'dark';
  children?: React.ReactNode;
}

export const ThemeProvider: React.FC<Props> = ({
  theme = 'light',
  children,
}) => (
  <StyledThemeProvider theme={React.useMemo(() => getTheme(theme), [theme])}>
    {children}
  </StyledThemeProvider>
);
