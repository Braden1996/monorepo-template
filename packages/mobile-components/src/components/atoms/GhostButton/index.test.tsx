import { render } from '@testing-library/react-native';
import * as React from 'react';

import { ThemeProvider } from '../../../theme';

import GhostButton from './index';

describe('<GhostButton />', () => {
  it('has 1 child', () => {
    const { container } = render(
      <ThemeProvider>
        <GhostButton>Hello</GhostButton>
      </ThemeProvider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
