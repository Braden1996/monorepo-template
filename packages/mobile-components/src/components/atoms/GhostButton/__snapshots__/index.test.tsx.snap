// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GhostButton /> has 1 child 1`] = `undefined`;
